sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/MetadataColumn.java

```diff
@@ -33,6 +33,8 @@
  * example, a partition value produced by bucket(id, 16) could be exposed by a metadata column. In
  * this case, {@link #transform()} should return a non-null {@link Transform} that produced the
  * metadata column's values.
+ *
+ * @since 3.1.0
```

Comment:
should this go to branch 3.1?

Reply:
@dongjoon-hyun has already merged this, see https://github.com/apache/spark/commit/24fc9f0492c1a49435e23fc265af53ed1bf234a8